.github/workflows/pr-build.yml

```diff
@@ -1,45 +1,122 @@
+          gh repo clone ${{ github.repository }}
+          cd revanced-manager
+          gh repo set-default ${{ github.repository }}
+          gh pr checkout ${{ inputs.pr-number }}
+          
+          echo "DATETIME=$( TZ='UTC+0' date --rfc-email )" >> $GITHUB_ENV
+          echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          # Make sure the release step uses its own credentials:
-          # https://github.com/cycjimmy/semantic-release-action#private-packages
           persist-credentials: false
-          fetch-depth: 0
+          
       - name: Setup JDK
         uses: actions/setup-java@v3
         with:
           java-version: '11'
           distribution: 'zulu'
+          cache: gradle
+          
       - name: Setup Flutter
         uses: subosito/flutter-action@v2
         with:
-          channel: 'stable'
-          cache: true
+          channel: ${{ inputs.flutter-branch }}
+          cache: ${{ inputs.flutter-cache }}
+          
       - name: Install Flutter dependencies
         run: flutter pub get
+        
       - name: Generate files with Builder
         run: flutter packages pub run build_runner build --delete-conflicting-outputs
+        
       - name: Build with Flutter
-        env:
+        continue-on-error: true
+        id: flutter-build
+        env: 
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: flutter build apk --debug
+        run: |
+          flutter build apk --${{ inputs.app-flavour }};
+          
+      - name: Prepare to comment
+        run: |
+          if [[ "${{ steps.flutter-build.outcome }}" == "success" ]]; then
+            echo "MESSAGE=✅ The CI has sucessfully built ReVanced Manager at ${{ env.COMMIT_HASH }}." >> $GITHUB_ENV
+          else
+            echo "MESSAGE=🚫 The CI has unable to build ReVanced Manager at ${{ env.COMMIT_HASH }}." >> $GITHUB_ENV
+          fi
+          
+      - name: "Comment to Pull Request #${{ inputs.pr-number }}"
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          GITHUB_TOKEN: ${{ github.token }}
+          pr_number: ${{ inputs.pr-number }}
+          mode: recreate
+          message: |
+            ## ⚒️ ReVanced PR Build workflow
+            
+            ${{ env.MESSAGE }}
+            
+            You can check the details on run ID [${{ github.run_id }}](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})!
+
+            ### ⚙️ Overview
+            - App flavor: ${{ inputs.app-flavour }}
+            - Flutter branch: ${{ inputs.flutter-branch }}
+            - Flutter cache: ${{ inputs.flutter-cache }}
+            - Start time: ${{ env.DATETIME }}
+
       - name: Upload build
         uses: actions/upload-artifact@v3
         with:
-          name: revanced-manager
-          path: build/app/outputs/flutter-apk/app-debug.apk
+          error: true
```

Comment:
Ensure that the CI report error when build failed, instead of success because we continue on error

Reply:
```suggestion
          if-no-files-found: error
```